agent/handlers/agentapi/taskprotection/v1/handlers/handlers.go

```diff
@@ -222,12 +221,54 @@ func GetTaskProtectionHandler(state dockerstate.TaskEngineState,
 			return
 		}
 
-		// TODO: Call ECS
 		logger.Info("GetTaskProtection endpoint was called", logger.Fields{
 			loggerfield.Cluster: cluster,
 			loggerfield.TaskARN: task.Arn,
 		})
-		writeJSONResponse(w, http.StatusOK, "Ok", getTaskProtectionRequestType)
+
+		ecsClient, responseCode, err := factory.newTaskProtectionClient(credentialsManager, task)
+		if err != nil {
+			writeJSONResponse(w, responseCode, err.Error(), getTaskProtectionRequestType)
+			return
+		}
+
+		response, err := ecsClient.GetTaskProtection(&ecs.GetTaskProtectionInput{
+			Cluster: aws.String(cluster),
+			Tasks:   aws.StringSlice([]string{task.Arn}),
+		})
+
+		if err != nil {
+			exceptionType, statusCode := getExceptionTypeAndStatusCode(err)
+			logger.Error("ECS throws an exception.", logger.Fields{
+				"ExceptionType":   exceptionType,
+				loggerfield.Error: err,
+			})
+			writeJSONResponse(w, statusCode, err.Error(), getTaskProtectionRequestType)
+			return
+		}
+
+		logger.Debug("getTaskProtection response:", logger.Fields{
+			loggerfield.TaskProtection: response.ProtectedTasks,
+			loggerfield.Reason:         response.Failures,
+		})
+
+		// there are no exceptions but there are failures when getting protection in scheduler
+		if len(response.Failures) > 0 {
+			writeJSONResponse(w, http.StatusBadRequest, response.Failures[0], getTaskProtectionRequestType)
+			return
+		}
+
+		// there are no exceptions but there are failures when setting protection in scheduler
```

Comment:
Is the comment okay? or are we supposed to write http.StatusOK when there are failures?

Reply:
This block (till line 265) should be removed. Seems to be an issue from rebase